common/flatpak-utils.c

```diff
@@ -3207,6 +3207,141 @@ flatpak_mtree_create_root (OstreeRepo        *repo,
   return TRUE;
 }
 
+static gboolean
+validate_mode (guint32 mode,
+               GError **error)
+{
+  if (mode & S_ISUID)
+    return glnx_throw (error, "suid");
+  if (mode & S_ISGID)
+    return glnx_throw (error, "sgid");
+  if (mode & S_IWOTH)
+    return glnx_throw (error, "world-writable");
+  return TRUE;
+}
+
+static gboolean
+check_commit_content_perms_recurse (OstreeRepo                     *repo,
+                                    OstreeRepoCommitTraverseIter   *iter,
+                                    GCancellable                   *cancellable,
+                                    GError                        **error)
+{
+  while (TRUE)
+    {
+      OstreeRepoCommitIterResult iterres =
+        ostree_repo_commit_traverse_iter_next (iter, cancellable, error);
+      g_autoptr(GError) local_error = NULL;
+
+      if (iterres == OSTREE_REPO_COMMIT_ITER_RESULT_ERROR)
+        return FALSE;
+      else if (iterres == OSTREE_REPO_COMMIT_ITER_RESULT_END)
+        break;
+      else if (iterres == OSTREE_REPO_COMMIT_ITER_RESULT_FILE)
+        {
+          char *name;
+          char *checksum;
+
+          ostree_repo_commit_traverse_iter_get_file (iter, &name, &checksum);
+
+          g_autoptr(GFileInfo) finfo = NULL;
+          g_autoptr(GVariant) xattrs = NULL;
+          if (!ostree_repo_load_file (repo, checksum, NULL, &finfo, &xattrs,
```

Comment:
Does this actually work? I have had issues with ostree_repo_load_file() not handling objects in the stage directory.
Looking at query_info_for_bare_content_object(), it does not seem to look at commit_stagedir_fd.